### ISSUE TITLE
Users should be able to indicate allowed methods for `unwrap_used`

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#15944]
Mostly because of `std::sync::Mutex` but I am not aware of any other use-case. It is also possible to directly whitelist the `lock` method in the lint without any user-provided data.

```rust
use std::sync::Mutex;

#[deny(clippy::unwrap_used)]
pub fn from_days(mutex: &Mutex<i32>) -> i32 {
    *mutex.lock().unwrap()
}
```

```
allowed-unwrap-used-functions = ["std::sync::Mutex::lock"]
```
